### ISSUE TITLE
[bug] Syntax highlighting #defines

### ISSUE THREAD

**Issue by ryanmjacobs** [neovim/neovim#1999]
This is my [.nvimrc](https://github.com/ryanmjacobs/ryans_dotfiles/blob/master/.vimrc) and my [.nvim/](https://github.com/ryanmjacobs/ryans_dotfiles/tree/master/.vim). It seems that everything after the `0` in a `#define` is interpreted as a comment. Works just fine in vim. Btw, I compile nvim from source at fd636fc.

![nvim-vs-vim](https://cloud.githubusercontent.com/assets/4251257/6223259/c08c73de-b619-11e4-974e-fe23fa3c1760.png)
![nvim-vs-vim-2](https://cloud.githubusercontent.com/assets/4251257/6223264/c62be158-b619-11e4-97c0-b57865a6f8d8.png)

(I apologize for the crude mouse drawings :grimacing:)

**Comment by justinmk** [comment]
could be a missing runtime update, or perhaps a regression in one of the recent runtime updates. What version of vim?

**Comment by justinmk** [comment]
Also, what happens with `nvim -u NONE` ?

**Comment by ryanmjacobs** [comment]
Vim's version:
```
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Feb  4 2015 08:02:36)
Included patches: 1-617
...
```

Neovim's version:
```
NVIM 0.0.0-alpha+201502162342 (compiled Feb 16 2015 20:39:01)
Commit: 0429857689ba98356bc80d01ebd540fe861e8db2
Build type: Debug
...
```

Running `nvim -u NONE` has no syntax highlighting, until I run `:syntax on`. Then, it still doesn't work.
![scrot](https://cloud.githubusercontent.com/assets/4251257/6223626/e0835b7c-b622-11e4-8411-496d41a43ef7.png)

**Comment by ryanmjacobs** [comment]
@justinmk Does the highlighting work correctly for you?

**Comment by justinmk** [comment]
No. I didn't try before because there was no sample file, but turns out this is enough to reproduce it:

    #define C0 16.35

So, it could be a missing runtime patch. We are a few behind.

**Comment by chrisbra** [comment]
Am 2015-02-17 06:39, schrieb Justin M. Keyes:
> No. I didn't try before because there was no sample file, but turns
> out this is enough to reproduce it:
> 
> #define C0 16.35
> 
> So, it could be a missing runtime patch. We are a few behind.

Yes, this looks like issue 
[281](http://code.google.com/p/vim/issues/detail?id=281)
from the Vim bug tracker.

Best,
Christian

**Comment by justinmk** [comment]
@chrisbra Thanks!

**Comment by iperry** [comment]
Should be resolved with this PR:
https://github.com/neovim/neovim/pull/2659

This one has been driving me insane.

**Comment by Pyrohh** [comment]
Closed in 04a3d5ecc7c9f626b28f60fdd76f08e0e9c867d3
